lib/ecto/schema.ex

```diff
@@ -252,7 +252,7 @@ defmodule Ecto.Schema do
   defined so structs and changeset functionalities are available.
   """
 
-  @type t :: %{__struct__: atom}
```

Comment:
`struct` is defined as `%{:__struct__ => atom, optional(atom) => any}`. It's because of a change in behaviour in OTP19 where `%{__struct__: atom}` only allows a single key in the map.